plugins/catalog-backend/src/service/DefaultEntitiesCatalog.ts

```diff
@@ -210,7 +210,14 @@ export class DefaultEntitiesCatalog implements EntitiesCatalog {
       };
     }
 
-    let entities: Entity[] = rows.map(e => JSON.parse(e.final_entity!));
+    let entities: Entity[] = rows.map(e => {
+      const entityJson = JSON.parse(e.final_entity!);
+      if (e.last_updated_at) {
+        entityJson.metadata.annotations['backstage.io/last_updated_at'] =
```

Comment:
This is the part of the pull request that I'll push back against. Modifying the data on read is something we've built towards avoiding.

The only place where we might insert data dynamically like this, would be inside the Stitcher (as it does for the orphan marker). Otherwise it'll be suspiciously absent from search.

But the thing is, if we do this, it'll become a permanent part of the contract. People will start relying on it. And I'm not sure we're ready for that without more thought. Could we make the pull request just do the database updates, but not expose it as part of the entity for now? Then we could consider how to get the data out in a future PR instead.

Reply:
thanks for advice. It's changed to expect the value of last_updated_at to be not null.